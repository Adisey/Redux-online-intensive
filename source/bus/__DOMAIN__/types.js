export const type = {
    TYPE:        'TYPE',
};
