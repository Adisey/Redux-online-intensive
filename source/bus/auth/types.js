export const type = {
    // Sync
    AUTHENTICATE:       'AUTHENTICATE',
    // Async
    SIGNUP_ASYNC:       'SIGNUP_ASYNC',
    LOGIN_ASYNC:        'LOGIN_ASYNC',
    AUTHENTICATE_ASYNC: 'AUTHENTICATE_ASYNC',
};
