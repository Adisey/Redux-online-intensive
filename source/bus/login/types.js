export const type = {
    // Sync
    AUTHENTICATEL: 'AUTHENTICATEL',
    // Async
    LOGIN_ASYNC: '',
};