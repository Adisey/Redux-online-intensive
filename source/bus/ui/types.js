export const type = {
    START_FETCHING: 'START_FETCHING',
    STOP_FETCHING:  'STOP_FETCHING',
};
