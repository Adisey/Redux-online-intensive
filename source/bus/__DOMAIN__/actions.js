import { type } from "./types";

export const postsActions ={
    action: () => {
        return {
            type:    type.TYPE,
        };
    },
};
