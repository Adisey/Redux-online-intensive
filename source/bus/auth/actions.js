import { type } from "./types";

export const authAction ={
    //Sync
    authenticate: () => {
        return {
            type: type.AUTHENTICATE,
        };
    },
    // Async
    signupAsync: (userData) => {
        return {
            type:    type.SIGNUP_ASYNC,
            payload: userData,
        };
    },
    loginAsync: (userData) => {
        console.log(`userData ->`, userData);
        return {
            type:    type.LOGIN_ASYNC,
            payload: userData,
        };
    },
};
