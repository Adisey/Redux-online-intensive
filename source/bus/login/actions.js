import { type } from "./types";

export const loginActions ={
    //Sync
    authenticatel: () => {
        return {
            type: type.AUTHENTICATEL,
        };
    },
    // Async
    loginAsync: (userData) => {
        console.log(`userrData ->`, userData);
        return {
            type:    type.LOGIN_ASYNC,
            payload: userData,
        };
    },
};

