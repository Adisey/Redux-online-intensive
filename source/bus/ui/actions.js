import { type } from "./types";

export const uiActions ={
    startFetching: () => {
        return {
            type: type.START_FETCHING,
        };
    },
    stopFetching: () => {
        return {
            type: type.STOP_FETCHING,
        };
    },
};
