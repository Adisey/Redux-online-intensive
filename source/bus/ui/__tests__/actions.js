/**
 * Created by PhpStorm
 * Project p501-redux-online-intensive
 * User: Adisey
 * Date: 28.08.2018
 * Time: 17:10
 */

// Actions
import { uiActions } from '../actions';

// Types
import { type } from '../types';

describe('Экшены UI:', () => {
    test('startFetching', () => {
        expect(uiActions.startFetching())
            .toEqual({
                type: type.START_FETCHING,
            });
    });

    test('stopFetching', () => {
        expect(uiActions.stopFetching())
            .toEqual({
                type: type.STOP_FETCHING,
            });
    });

    test('setOnlineState', () => {
        expect(uiActions.setOnlineState())
            .toEqual({
                type: type.SET_ONLINE_STATE,
            });
    });

    test('setOfflineState', () => {
        expect(uiActions.setOfflineState())
            .toEqual({
                type: type.SET_OFFLINE_STATE,
            });
    });

    test('emitError', () => {
        expect(uiActions.emitError(__.error, __.errorMessage))
            .toEqual({
                type:    type.EMIT_ERROR,
                payload: __.error,
                meta:    __.errorMessage,
            });
    });

    test('emitError - meta=null', () => {
        expect(uiActions.emitError(__.error))
            .toEqual({
                type:    type.EMIT_ERROR,
                payload: __.error,
                meta:    null,
            });
    });
});

describe('Экшены UI Snapshot:', () => {
    test('startFetching Snapshot', () => {
        expect(uiActions.startFetching())
            .toMatchSnapshot();
    });

    test('stopFetching Snapshot', () => {
        expect(uiActions.stopFetching())
            .toMatchSnapshot();
    });

    test('setOnlineState Snapshot', () => {
        expect(uiActions.setOnlineState())
            .toMatchSnapshot();
    });

    test('setOfflineState Snapshot', () => {
        expect(uiActions.setOfflineState())
            .toMatchSnapshot();
    });

    test('emitError Snapshot', () => {
        expect(uiActions.emitError(__.error, __.errorMessage))
            .toMatchSnapshot();
    });

    test('emitError - meta=null Snapshot', () => {
        expect(uiActions.emitError(__.error))
            .toMatchSnapshot();
    });
});

