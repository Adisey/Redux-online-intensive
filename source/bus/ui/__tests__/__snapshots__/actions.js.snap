// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Экшены UI Snapshot: emitError - meta=null Snapshot 1`] = `
Object {
  "meta": null,
  "payload": [Error: TEST_ERROR_MESSAGE.],
  "type": "EMIT_ERROR",
}
`;

exports[`Экшены UI Snapshot: emitError Snapshot 1`] = `
Object {
  "meta": "TEST_ERROR_MESSAGE.",
  "payload": [Error: TEST_ERROR_MESSAGE.],
  "type": "EMIT_ERROR",
}
`;

exports[`Экшены UI Snapshot: setOfflineState Snapshot 1`] = `
Object {
  "type": "SET_OFFLINE_STATE",
}
`;

exports[`Экшены UI Snapshot: setOnlineState Snapshot 1`] = `
Object {
  "type": "SET_ONLINE_STATE",
}
`;

exports[`Экшены UI Snapshot: startFetching Snapshot 1`] = `
Object {
  "type": "START_FETCHING",
}
`;

exports[`Экшены UI Snapshot: stopFetching Snapshot 1`] = `
Object {
  "type": "STOP_FETCHING",
}
`;
